File: src/courses/data/catalog/02-dsa/advanced-dsa/implementations/javascript/scc.js
/**
 * JavaScript Implementation: Strongly Connected Components (Tarjan & Kosaraju)
 * Invariant: O(V + E) single or double DFS passes
 */
export class StronglyConnectedComponentsTarjanKosaraju {
  constructor() {
    this.items = [];
  }

  // TODO: Implement verified invariant
  insert(val) {
    this.items.push(val);
  }

  get size() {
    return this.items.length;
  }
}

File: src/courses/data/catalog/02-dsa/advanced-dsa/implementations/java/scc.java
/**
 * Java Implementation: Strongly Connected Components (Tarjan & Kosaraju)
 * Invariant: O(V + E) single or double DFS passes
 */
package seed.dsa.core;

import java.util.*;

public class StronglyConnectedComponentsTarjanKosaraju<T> {
    private final List<T> items = new ArrayList<>();

    // TODO: Implement your solution here
    public void add(T value) {
        items.add(value);
    }

    public int size() {
        return items.size();
    }
}
